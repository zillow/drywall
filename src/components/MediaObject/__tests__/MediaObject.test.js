import React from 'react';
import TestRenderer from 'react-test-renderer';
import { isStyledComponent } from 'styled-components';
import { MediaObject } from '../../../index';

describe('<MediaObject>', () => {
    it('can render without theme', () => {
        const testRenderer = TestRenderer.create(<MediaObject media="media">body</MediaObject>);
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it('is a styled-component', () => {
        expect(isStyledComponent(MediaObject)).toBe(true);
    });

    it('does not pass "media" prop through to DOM', () => {
        const testRenderer = TestRenderer.create(<MediaObject media="media">body</MediaObject>);
        const testInstance = testRenderer.root;
        const mediaObject = testInstance.findByType('div');
        expect(mediaObject.props.media).toBeUndefined();
    });

    it('does not pass "direction" prop through to DOM', () => {
        const testRenderer = TestRenderer.create(
            <MediaObject media="media" direction="column">
                body
            </MediaObject>
        );
        const testInstance = testRenderer.root;
        const mediaObject = testInstance.findByType('div');
        expect(mediaObject.props.direction).toBeUndefined();
    });

    it('does not pass unrecognized react props through to DOM', () => {
        const testRenderer = TestRenderer.create(
            <MediaObject media="media" data-foobar="baz" fooBar="baz">
                body
            </MediaObject>
        );
        const testInstance = testRenderer.root;
        const mediaObject = testInstance.findByType('div');
        expect(mediaObject.props['data-foobar']).toBe('baz');
        expect(mediaObject.props.fooBar).toBeUndefined();
    });

    it('customizes the layout with renderLayout render function', () => {
        const CustomLayout = () => <p>custom layout</p>;
        const testRenderer = TestRenderer.create(
            <MediaObject renderLayout={() => <CustomLayout />} />
        );
        const testInstance = testRenderer.root;
        expect(() => testInstance.findByType(CustomLayout)).not.toThrow();
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it('customizes the layout with renderLayout node', () => {
        const CustomLayout = () => <p>custom layout</p>;
        const testRenderer = TestRenderer.create(<MediaObject renderLayout={<CustomLayout />} />);
        const testInstance = testRenderer.root;
        expect(() => testInstance.findByType(CustomLayout)).not.toThrow();
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });
});
