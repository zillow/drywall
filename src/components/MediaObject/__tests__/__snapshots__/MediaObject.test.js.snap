// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaObject> can render without theme 1`] = `
<div
  className="MediaObject__StyledMediaObject-sc-12gs3hz-0 MediaObject-sc-12gs3hz-1 "
>
  <div
    className="MediaObject__Media-sc-12gs3hz-3 "
  >
    media
  </div>
  <div
    className="MediaObject__Body-sc-12gs3hz-2 "
  >
    body
  </div>
</div>
`;

exports[`<MediaObject> customizes the layout with renderLayout node 1`] = `
<div
  className="MediaObject__StyledMediaObject-sc-12gs3hz-0 MediaObject-sc-12gs3hz-1 "
>
  <p>
    custom layout
  </p>
</div>
`;

exports[`<MediaObject> customizes the layout with renderLayout render function 1`] = `
<div
  className="MediaObject__StyledMediaObject-sc-12gs3hz-0 MediaObject-sc-12gs3hz-1 "
>
  <p>
    custom layout
  </p>
</div>
`;
